File: 1.1/src/cz/mg/vulkan/PFNvkCmdSetDepthBias.java
package cz.mg.vulkan;

public class PFNvkCmdSetDepthBias extends VkFunctionPointer {
    public PFNvkCmdSetDepthBias() {
    }

    protected PFNvkCmdSetDepthBias(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdSetDepthBias(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdSetDepthBias(long value) {
        setValue(value);
    }

    public PFNvkCmdSetDepthBias(VkInstance instance) {
        super(instance, new VkString("vkCmdSetDepthBias"));
    }

    public void call(VkCommandBuffer commandBuffer, float depthBiasConstantFactor, float depthBiasClamp, float depthBiasSlopeFactor){
        callNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), depthBiasConstantFactor, depthBiasClamp, depthBiasSlopeFactor);
    }

    protected static native void callNative(long vkaddress, long commandBuffer, float depthBiasConstantFactor, float depthBiasClamp, float depthBiasSlopeFactor);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkCmdSetDepthBias.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdSetDepthBias_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jlong depthBiasConstantFactor, jlong depthBiasClamp, jlong depthBiasSlopeFactor){
    (void)env;
    (void)jc;
    PFN_vkCmdSetDepthBias f = (PFN_vkCmdSetDepthBias)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        *((float*)jniLongToPointer(depthBiasConstantFactor)),
        *((float*)jniLongToPointer(depthBiasClamp)),
        *((float*)jniLongToPointer(depthBiasSlopeFactor))
    );
}


void Java_cz_mg_vulkan_PFNvkCmdSetDepthBias_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jfloat depthBiasConstantFactor, jfloat depthBiasClamp, jfloat depthBiasSlopeFactor){
    (void)env;
    (void)jc;
    PFN_vkCmdSetDepthBias f = (PFN_vkCmdSetDepthBias)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        (float)depthBiasConstantFactor,
        (float)depthBiasClamp,
        (float)depthBiasSlopeFactor
    );
}

